src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl

```diff
@@ -0,0 +1,142 @@
+    γ = cp_d / cv_d
+
+    # TODO!: Replace with Thermodynamics.jl
+    pressure(ρθ) = ρθ >= 0 ? p_0 * (R_d * ρθ / p_0)^γ : NaN
+
+    # unity tensor for pressure term calculation 
+    # in horizontal spectral divergence
+    I = Ref(Geometry.Axis2Tensor(
+        (Geometry.Cartesian1Axis(), Geometry.Cartesian1Axis()),
+        @SMatrix [1.0]
+    ),)
+
+    # operators
+    # spectral horizontal operators
+    hdiv = Operators.Divergence()
+
+    # vertical FD operators with BC's
+    # interpolators
+    scalar_interp_c2f = Operators.InterpolateC2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+    vector_interp_c2f = Operators.InterpolateC2F(
+        bottom = Operators.SetValue(Geometry.Cartesian1Vector(0.0)),
+        top = Operators.SetValue(Geometry.Cartesian1Vector(0.0)),
+    )
+    tensor_interp_f2c = Operators.InterpolateF2C()
+
+    # gradients
+    scalar_grad_c2f = Operators.GradientC2F()
+    B = Operators.SetBoundaryOperator(
+        bottom = Operators.SetValue(Geometry.Cartesian3Vector(0.0)),
+        top = Operators.SetValue(Geometry.Cartesian3Vector(0.0)),
+    )
+
+    # divergences
+    vector_vdiv_f2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.Cartesian3Vector(0.0)),
+        top = Operators.SetValue(Geometry.Cartesian3Vector(0.0)),
+    )
+    tensor_vdiv_c2f = Operators.DivergenceC2F(
+        bottom = Operators.SetDivergence(Geometry.Cartesian3Vector(0.0)),
+        top = Operators.SetDivergence(Geometry.Cartesian3Vector(0.0)),
+    )
+    tensor_vdiv_f2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(
+            Geometry.Cartesian3Vector(0.0) ⊗ Geometry.Cartesian1Vector(0.0),
+        ),
+        top = Operators.SetValue(
+            Geometry.Cartesian3Vector(0.0) ⊗ Geometry.Cartesian1Vector(0.0),
+        ),
+    )
+
+    function rhs!(dY, Y, Ya, t)
+        dYm = dY.nhm
+        dρ = dYm.ρ
+        dρuh = dYm.ρuh
+        dρw = dYm.ρw
+        dρθ = dYm.ρθ
+        Ym = Y.nhm
+        ρ = Ym.ρ
+        ρuh = Ym.ρuh
+        ρw = Ym.ρw
+        ρθ = Ym.ρθ
+
+        # advection, pressure gradients, sources
+        # density
+        @. dρ = -hdiv(ρuh)
+        @. dρ -= vector_vdiv_f2c(ρw)
+
+        # potential temperature
+        @. dρθ = -hdiv(ρuh * ρθ / ρ)
+        @. dρθ -= vector_vdiv_f2c(ρw * scalar_interp_c2f(ρθ / ρ))
+
+        # horizontal momentum
+        @. dρuh = -hdiv(ρuh ⊗ ρuh / ρ + pressure(ρθ) * I)
+        @. dρuh -= tensor_vdiv_f2c(ρw ⊗ vector_interp_c2f(ρuh / ρ))
+
+        # vertical momentum
+        # TODO: How ugly is this?! (-_-)' . ￣□￣｜｜  (ಥ﹏ಥ)
+        uh_f = @. vector_interp_c2f(ρuh / ρ)
+        @. dρw = hdiv(uh_f ⊗ ρw)
+        @. dρw += B(
+            Geometry.transform(
+                Geometry.Cartesian3Axis(),
+                -(scalar_grad_c2f(pressure(ρθ))) +
+                scalar_interp_c2f(ρ) * Geometry.Covariant3Vector(-g), # TODO!: Not generally a Covariant3Vector
```

Comment:
you could use `grad(\Phi(z))` here to get the right output

Reply:
@jakebolewski  I guess you meant something like this line [here](https://github.com/CliMA/ClimaCore.jl/blob/1a3c9463d367aa4dede9a11ffd81249ea8fa9017/examples/hybrid/bubble.jl#L194).
It is a bit hard to access the coordinates and this is only a hacky way of doing it for now...
Do you think if there's a simple way to access `coords.z` in the way the model is currently written?

Reply:
yes, it is fine for now as a hacky soln but I think eventually you'll want to be able to grab references to the coordnate data (I assume mostly z) so it is worth thinking about how best to pass that information through.  Accessing it as a field is just a passed reference to a view so it's really cheap.

Reply:
Sounds good!